polyfill/lib/ecmascript.mjs

```diff
@@ -31,7 +31,20 @@ export const ES = ObjectAssign(ObjectAssign(ObjectAssign({}, Cast), ES2019), {
   GetIntrinsic: (intrinsic) => {
     return intrinsic in INTRINSICS ? INTRINSICS[intrinsic] : GetIntrinsic(intrinsic);
   },
-
+  ValidPropertyBag: (bag, anyof = [], noneof = []) => {
+    if (!bag || 'object' !== typeof bag) return false;
+    let any = !anyof.length;
+    for (let prop of anyof) {
+      const value = ES.ToNumber(bag[prop]);
+      any = any || Number.isFinite(value);
+    }
+    let none = false;
+    for (let prop of noneof) {
+      const value = ES.ToNumber(bag[prop]);
+      none = none || (Number.isFinite(value) && !!value);
+    }
```

Comment:
I think the entire concept of "this object is not allowed to have these keys" is an anti-pattern. Does it exist anywhere else in the platform?

It's especially weird because you're allowing `{foo: NaN}` and `{foo: 0}` even when `foo` is forbidden.